### DISCUSSION TITLE
Java API for interpreting real expr ?

### DISCUSSION THREAD

**Discussion by classicwuhao** [Z3Prover/z3#6019] [Q&A] [answered]
I am wondering whether Z3 java API provides a way of interpreting a real expression? Or I have to cast a real expression (a/b) externally myself?

**Comment by NikolajBjorner** [comment, marked as answer]
do you mean get a double or float approximation?

**Reply by classicwuhao** [reply]
I think I find the solution. I cast an expr to a RatNum and then get its float approximation, I don't know if this is the correct way of doing it.

**Reply by NikolajBjorner** [reply]
this is the current approach yes.

**Reply by classicwuhao** [reply]
Thanks.
